execution/aggregate/hashaggregate.go

```diff
@@ -125,20 +120,15 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 		return nil, err
 	}
 
-	if a.paramOp != nil {
-		args, err := a.paramOp.Next(ctx)
-		if err != nil {
-			return nil, err
-		}
-		for i := range a.params {
-			a.params[i] = math.NaN()
-			if i < len(args) && len(args[i].Samples) > 0 {
-				a.params[i] = args[i].Samples[0]
-				a.paramOp.GetPool().PutStepVector(args[i])
-			}
-		}
-		a.paramOp.GetPool().PutVectors(args)
+	args, err := a.paramOp.Next(ctx)
```

Comment:
Would this cause extra allocations for aggregations without a parameter?

Reply:
Didnt see a difference in benchmarks!

Reply:
Ill resolve this really quick